### PR TITLE
Prevent redundant autoscaling events on the autoscaler

### PR DESCRIPTION
**Pull request by RobertLucian**
The autoscaling events were actually a consequence of the autoscaler comparing the request count (in our case `min_replicas`) with the number of pods created by k8s until that moment. When there are thousands of pods that need to be created, the control plane needs more than a few minutes to reach its target number.

```
{"level":"info","ts":1626385146.2022922,"caller":"autoscaler/main.go:230","message":"Starting autoscaler server on :8000"}
{"level":"info","ts":1626385188.0080802,"caller":"autoscaler/autoscaler.go:175","message":"autoscaler init","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385248.422167,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 500 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385258.5042984,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 500 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385268.6129313,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1000 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385278.820514,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1000 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385288.90154,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1500 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385299.00116,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1500 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385309.0716608,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1500 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
{"level":"info","ts":1626385319.223561,"caller":"autoscaler/autoscaler.go:293","message":"autoscaling event: 1999 -> 10000","apiName":"hello-world","apiKind":"RealtimeAPI"}
rpc error: code = Unknown desc = Error: No such container: e9bf5b57424c783027a924644eee033e02e11bb4dff1c4ae3a26ef6539a9b436
```

---

checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

File: pkg/autoscaler/async_scaler.go [hunk 1]
```diff
@@ -126,5 +126,5 @@ func (s *AsyncScaler) CurrentReplicas(apiName string) (int32, error) {
 		return 0, errors.ErrorUnexpected("unable to find k8s deployment", apiName)
```
Comment (deliahu) [inline pkg/autoscaler/async_scaler.go:126]: nit: rename `CurrentReplicas` to `CurrentRequestedReplicas`

**Review by deliahu — APPROVED** [review]

File: pkg/autoscaler/async_scaler.go (hunk 1 above)
Comment (miguelvr) [reply pkg/autoscaler/async_scaler.go:126]: the method is satisfying an interface

File: pkg/autoscaler/async_scaler.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (s *AsyncScaler) Scale(apiName string, request int32) error {
 		return errors.ErrorUnexpected("unable to find k8s deployment", apiName)
 	}
 
-	current := deployment.Status.Replicas
+	current := *deployment.Spec.Replicas
```
Comment (miguelvr) [inline pkg/autoscaler/async_scaler.go:55]: although it's probably not going to happen, we should check if the `Spec.Replicas` is nil to avoid panics

Comment (miguelvr) [reply pkg/autoscaler/async_scaler.go:55]: same applies to the rest of the PR

**Review by miguelvr — APPROVED** [review]

**Pull request merged**